### PR TITLE
Disable setting remediationAction and severity for Gatekeeper policy templates

### PR DESCRIPTION
**Pull request by mprahl**
This disables the remediationAction and severity options in the policy templates section of the policy wizard when it is a Gatekeeper constraint or ConstraintTemplate.

Instead, text is displayed that tells the user to use the YAML editor to make modifications.

Relates:
https://issues.redhat.com/browse/ACM-4808

image

### CODE REVIEW

**Comment by mprahl** [comment]
@dockerymick could you please review the new text of "Gatekeeper policy templates must be customized through the YAML editor."?

/cc @dhaiducek @JustinKuli @yiraeChristineKim

File: frontend/src/resources/policy.ts [hunk 1]
```diff
@@ -55,8 +55,8 @@ export interface PolicyTemplate {
           subjects?: ResourceRef[]
         }
       }[]
-      remediationAction: string
-      severity: string
+      remediationAction?: string
```
Comment (dhaiducek) [inline frontend/src/resources/policy.ts:58]: Current linting error:
```
error TS2322: Type 'string | undefined' is not assignable to type 'string'.
Type 'undefined' is not assignable to type 'string'. 
```
It looks like having `remediationAction` optional is causing this assignment to be upset?
https://github.com/stolostron/console/blob/9f5340910630f9541ed0ae69c90c29c0dfa18f98/frontend/src/routes/Governance/common/util.tsx#L484

**Review by dhaiducek — CHANGES_REQUESTED** [review]

File: frontend/public/locales/en/translation.json [hunk 2]
```diff
@@ -1369,6 +1369,7 @@
   "formerror.valid.name": "Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character",
   "G3 - NVIDIA Tesla M60 GPUs": "G3 - NVIDIA Tesla M60 GPUs",
   "G4 - NVIDIA T4 Tensor Core GPUs": "G4 - NVIDIA T4 Tensor Core GPUs",
+  "Gatekeeper policy templates must be customized through the YAML editor.": "Gatekeeper policy templates must be customized through the YAML editor.",
```
Comment (dockerymick) [inline frontend/public/locales/en/translation.json:1372]:
```suggestion
  "Gatekeeper policy templates must be customized through the YAML editor.": "Gatekeeper policy templates must be customized using the YAML editor.",
```

File: frontend/src/wizards/Governance/Policy/PolicyWizard.tsx [hunk 3]
```diff
@@ -469,34 +469,42 @@ export function PolicyWizardTemplates(props: { policies: IResource[] }) {
...
-            { label: t('medium'), value: 'medium' },
-            { label: t('high'), value: 'high' },
-          ]}
-          required
-        />
+        </WizHidden>
+
+        <WizHidden hidden={(template: any) => !template?.objectDefinition?.apiVersion?.includes('gatekeeper.sh/')}>
+          <TextContent>
+            <Text>{t('Gatekeeper policy templates must be customized through the YAML editor.')}</Text>
```
Comment (dockerymick) [inline frontend/src/wizards/Governance/Policy/PolicyWizard.tsx:505]:
```suggestion
            <Text>{t('Gatekeeper policy templates must be customized using the YAML editor.')}</Text>
```

File: frontend/src/wizards/Governance/Policy/policyWizard.test.tsx [hunk 4]
```diff
@@ -61,4 +101,13 @@ describe('Policy wizard', () => {
 
     expect(screen.getByRole('checkbox', { name: /cluster-set-01/i })).not.toBeNull()
   })
+
+  test('policy template customization is disabled for Gatekeeper policy', async () => {
+    const { container } = render(<TestPolicyWizardGK />)
+    screen.getByRole('button', { name: /policy templates/i }).click()
+
+    await waitForText('Gatekeeper policy templates must be customized through the YAML editor.', true)
```
Comment (dockerymick) [inline frontend/src/wizards/Governance/Policy/policyWizard.test.tsx:109]:
```suggestion
    await waitForText('Gatekeeper policy templates must be customized using the YAML editor.', true)
```

**Review by dockerymick — APPROVED** [review]
Made a few suggestions

**Review by dockerymick — APPROVED** [review]
Looks good to me

**Comment by mprahl** [comment]
@KevinFCormier could you please "approve" this PR?

**Pull request merged**